rfd/0001-rfd.md

```diff
@@ -0,0 +1,385 @@
+   actively discussed
+
+3. `published` - Once (or if) discussion has converged and the Pull Request is
+   ready to be merged, it should be updated to the published state before
+   merge. Note that just because something is in the published state does not
+   mean that it cannot be updated and corrected.
+
+4. `implemented` - Once an idea has been entirely implemented, it should be in
+   the implemented state. Comments on ideas in the implemented state should
+   generally be raised as issues — but if the comment represents a call for a
+   significant divergence from or extension to committed functionality, a new
+   RFD may be called for.
+
+5. `abandoned` - Finally, if an idea is found to be non-viable (that is,
+   deliberately never implemented) or if an RFD should be otherwise indicated
+   that it should be ignored, it can be moved into the abandoned state.
+
+##### Content
+
+Generally the contain of the RFD would consist of the following sections:
+
+1. **Introduction** - A brief introduction to the RFD and the context
+   in which it is being proposed.
+
+2. **Proposal** - The proposal section should contain the main content of the
+   RFD. This is where the idea is explained in detail.
+
+   Your proposal ideally should include the following points. However, use
+   your best judgment to determine what is most relevant and necessary for
+   your specific proposal:
+
+   - **Objective and goals:** clearly state the goal or problem that the
+     proposal aims to address.
+
+   - **Scope:** Define the boundaries of the proposal, what is in scope and
+     what is out of scope.
+
+   - **Multiple viable options:** Present various options to address the
+     issue, including the benefits and drawbacks of each option.
+
+   - **Reasoning and Data**: Provide detailed reasoning for the proposed
+     options, supported by data and references wherever possible.
+
+   - **Future Considerations**: Outline any future work or considerations that
+     might be relevant once the proposal is implemented.
+
+3. **Footnotes** - Footnotes should be used to provide references to external
+   sources or to provide additional information.
+
+We can leverage the Github flavored markdown for writing the contents of the RFD.
+Refer to the [Github Docs](https://docs.github.com/en/enterprise-cloud@latest/get-started/writing-on-github/basic-writing-and-formatting-syntax)
+for more information on the markdown syntax.
+
+##### Diagrams
+
+You're encouraged to include diagrams in the RFD. Diagrams can be used to
+illustrate the proposed changes or to provide additional context. Typically we
+would use [Mermaid](https://mermaid-js.github.io/mermaid/#/) for diagrams.
+
+```mermaid
+graph TD;
+  A-->B;
+  A-->C;
+  B-->D;
+  C-->D;
+```
+
+##### Assets
+
+If the RFD requires additional assets, such as images or diagrams, these should
+be stored in a separate directory within the repository. The `assets/`
```

Comment:
Why not in `rfd/assets/`? Imho it would be better to have assets for RFDs in the same folder rather than cross-referencing from the root directory.

Reply:
Yes you're correct, that was the intention. I'll make sure to update it to resemble that